src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/ListExtensions.cs

```diff
@@ -17,4 +18,11 @@ public static void SetCapacityIfLarger<T>(this List<T> list, int newCapacity)
             list.Capacity = newCapacity;
         }
     }
+
+    // On .NET Framework, List<T>.ToArray() will create a new empty array for any
+    // empty List<T>. This avoids that extra allocation.
+    public static T[] ToArrayOrEmpty<T>(this List<T>? list)
```

Comment:
Wrap this method in `#if !NETCOREAPP`?

Reply:
There's no NETCOREAPP API that works like this extension method does, since this can be called on a null `List<T>` and return an empty array.